From 250bee9e02546846d70dbe88ac36bb83c6a7ebdf Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sun, 22 Sep 2019 09:27:24 +0200
Subject: [PATCH] configure.ac: add --disable-doc option

Add --disable-doc option to allow the user to disable documentation

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Sent upstream: https://savannah.gnu.org/bugs/index.php?56931]
---
 Makefile.am  | 6 +++++-
 configure.ac | 7 +++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index dbb5719..2db307a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -18,6 +18,10 @@
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4
 
 EXTRA_DIST = configure TODO LICENSE gl/m4/gnulib-cache.m4
-SUBDIRS = gl src doc tests
+SUBDIRS = gl src tests
+
+if ENABLE_DOC
+SUBDIRS += doc
+endif
 
 gengetoptdoc_DATA = ChangeLog COPYING NEWS THANKS INSTALL README LICENSE
diff --git a/configure.ac b/configure.ac
index 0ee89db..915706c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -61,6 +61,13 @@ AC_ARG_ENABLE(
 	AC_COMPILE_WARNINGS
 )
 
+AC_ARG_ENABLE(
+	[doc],
+	[--enable-doc			enable documentation],
+	[enable_doc=$enableval],
+	[enable_doc=yes])
+AM_CONDITIONAL(ENABLE_DOC, test x"$enable_doc" = xyes)
+
 AC_ARG_WITH(
 	[gengen],
 	[  --with-gengen=mygengen       gengen to use],
-- 
2.23.0

